libs/eventually/src/app.ts

```diff
@@ -94,38 +94,38 @@ export abstract class AppBase extends Builder implements Reader {
   }
 
   /**
-   * Handles reducible storage
-   * @param handler the message handler
-   * @param callback the concrete message handling
-   * @param expectedVersion optional reducible expected version
-   * @param publishCallback optional callback to publish events with "at-least-once" delivery guarantees
-   * @returns the reduced snapshots
+   * Generic message handler
+   * @param handler Message handler
+   * @param callback Concrete message handling callback
+   * @param metadata Message metadata
+   * @param publishCallback Optional callback to publish events with "at-least-once" delivery guarantees
+   * @returns Reduced snapshots
    */
   private async _handle<M extends Payload, C, E>(
     handler: MessageHandler<M, C, E>,
     callback: (state: M) => Promise<Message<string, Payload>[]>,
     metadata: CommittedEventMetadata,
-    expectedVersion?: number,
     publishCallback?: (
       events: CommittedEvent<keyof E & string, Payload>[]
     ) => Promise<void>
   ): Promise<Snapshot<M>[]> {
     const streamable = getStreamable(handler);
     const reducible = getReducible(handler);
-    let { state, count } = reducible // eslint-disable-line prefer-const
+    const snapshot = reducible
       ? await this.load(reducible)
-      : { state: undefined, count: 0 };
-    const events = await callback(state);
+      : { event: undefined, count: 0 };
+    const events = await callback(snapshot.state);
     events.map((event) => this._validate(event));
     if (streamable) {
       const committed = await store().commit(
         streamable.stream(),
         events,
         metadata,
-        expectedVersion,
+        metadata.causation.command?.expectedVersion,
```

Comment:
what is the benefit of storing the expected version in the command causation?, if it got to the store, we can safely assume the expected version was right or not relevant... am I missing something?

Reply:
this is just to audit when the command was invoked with expectedVersion or not